### ISSUE TITLE
Failure: Invalid wire type

### ISSUE THREAD

**Issue by j-r79** [DrozmotiX/ioBroker.esphome#200]
I use some devices with esphome in iobroker which have all the same problem. For many days everything works without any problem, but then this message pops up every second in the log:

ESPHome client 192.168.100.145 Error: Failure: Invalid wire type: %s (at position %s)

I can fix the problem by removing the device in the instance settings (ip & delete device) and adding it again. Then it will work again for some days...

**Versions:**  
 - Adapter version: 0.5.0-beta.10
 - JS-Controller version: 5.0.17
 - Node version: v18.19.0
 - Operating system: raspibian 6.1.0-rpi7-rpi-v8 #1 SMP PREEMPT Debian 1:6.1.63-1+rpt1 (2023-11-24) aarch64 GNU/Linux

**Comment by j-r79** [comment]
If i press the update-button when the error is active, the adapter crashes:

`esphome.0
2024-03-01 07:45:10.553	error	sendMessage timeout waiting for HelloResponse
esphome.0
2024-03-01 07:45:10.553	error	Error: sendMessage timeout waiting for HelloResponse at Timeout.<anonymous> (/opt/iobroker/node_modules/@2colors/esphome-native-api/lib/connection.js:201:20) at listOnTimeout (node:internal/timers:569:17) at process.processTimers (node:internal/timers:512:7)
esphome.0
2024-03-01 07:45:10.552	error	unhandled promise rejection: sendMessage timeout waiting for HelloResponse
esphome.0
2024-03-01 07:45:10.551	error	Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch().

esphome.0
2024-03-01 07:45:05.572	error	ESPHome client 192.168.100.145 Error: Failure: Invalid wire type: %s (at position %s)`

After the crash the esphome-device is working again:

`ESPHome client 192.168.100.145 connected`

**Comment by DutchmanNL** [comment]
is this still happening and reproducible ? the error itself I cannot solve at its an dependency but the adapter crash should defiantly NOT happen and I have to troubleshoot that + implement proper error handling. but need someone who can test and reproduce :)

**Comment by j-r79** [comment]
Yes, it still happens. I can not force it but i wrote a script which reads the iobroker log an restarts the instance if the error is written into the log.

**Comment by DutchmanNL** [comment]
> Yes, it still happens. I can not force it but i wrote a script which reads the iobroker log an restarts the instance if the error is written into the log.

would it be possible to activate debug logging to track & trace it until the crash ? I would need the last 20 log entries before the crash

**Comment by SimonFischer04** [comment]
@j-r79 have you been able to get more logs about this yet?

**Comment by SimonFischer04** [comment]
@Apollon77 close stale?
